### PR TITLE
Feat: vela logs support multicluster

### PR DESCRIPTION
**Pull request by chivalryq**
### Description of your changes

`vela logs` are now available for multicluster.
image

Fixes #

I have:

- [x] Read and followed KubeVela's [contribution process](https://github.com/oam-dev/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/oam-dev/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### CODE REVIEW

**Review by Somefive — APPROVED** [review]
lgtm

File: pkg/utils/common/common.go [hunk 1]
```diff
@@ -239,6 +240,51 @@ func RealtimePrintCommandOutput(cmd *exec.Cmd, logFile string) error {
 	return nil
 }
 
+// ClusterObject2Map convert ClusterObjectReference to a readable map
+func ClusterObject2Map(refs ...common.ClusterObjectReference) map[string]string {
```
Comment (hongchaodeng) [inline pkg/utils/common/common.go:244]:
```suggestion
func clusterObject2Map(refs []common.ClusterObjectReference) map[string]string {
```

File: references/appfile/app.go [hunk 2]
```diff
@@ -84,7 +86,12 @@ func LoadApplication(namespace, appName string, c common.Args) (*v1beta1.Applica
 	return app, nil
 }
 
-// GetComponents will get oam components from Appfile.
+// GetAppliedResources get applied resources form v1beta1.Application.
+func GetAppliedResources(app *v1beta1.Application) []common2.ClusterObjectReference {
```
Comment (hongchaodeng) [inline references/appfile/app.go:90]: No need to add this function..

File: references/cli/logs.go [hunk 3]
```diff
@@ -41,51 +43,47 @@ import (
...
 }
 
 // Args creates arguments for `logs` command
 type Args struct {
-	Output string
-	Env    *types.EnvMeta
-	C      common.Args
-	App    *v1beta1.Application
+	Output    string
+	C         common.Args
```
Comment (hongchaodeng) [inline references/cli/logs.go:84]:
```suggestion
	Args         common.Args
```

**Review by hongchaodeng — APPROVED** [review]
lgtm in general
Please pay attention to naming...

File: pkg/utils/common/common.go [hunk 4]
```diff
@@ -239,6 +240,51 @@ func RealtimePrintCommandOutput(cmd *exec.Cmd, logFile string) error {
...
+	}
+	return objs
+}
+
+// AskToChooseOneAppliedResource will ask users to select one applied resource of the application if more than one
+// resources is a map for component to applied resources
+// return the selected ClusterObjectReference
+func AskToChooseOneAppliedResource(resources []common.ClusterObjectReference) (*common.ClusterObjectReference, error) {
+	if len(resources) == 0 {
+		return nil, fmt.Errorf("no applied resources exist in the application")
```
Comment (zzxwill) [inline pkg/utils/common/common.go:261]: What's an `applied resource`?

File: pkg/utils/common/common.go [hunk 5]
```diff
@@ -239,6 +240,51 @@ func RealtimePrintCommandOutput(cmd *exec.Cmd, logFile string) error {
...
+	}
+	if len(resources) == 1 {
+		return &resources[0], nil
+	}
+	opMap := ClusterObject2Map(resources...)
```
Comment (zzxwill) [inline pkg/utils/common/common.go:266]: Do you mean `obMap`?

File: pkg/utils/common/common.go (hunk 4 above)
Comment (chivalryq) [reply pkg/utils/common/common.go:261]: It's generally a combination of {cluster, gvk, namespace, name}

File: pkg/utils/common/common.go (hunk 5 above)
Comment (chivalryq) [reply pkg/utils/common/common.go:266]: I thought op is short for options...

**Pull request merged**